docs/src/references/agent/device-management-api.md

```diff
@@ -0,0 +1,330 @@
+- There only two terminal states: `"successful"` and `"failed"`.
+- The retained messages are finally cleared by the requester.
+
+```mermaid
+flowchart LR
+    init --> executing --> successful
+    executing --> failed
+    successful --> done
+    failed --> done
+
+    subgraph init_state["status: init"]
+      init(gives all the data to proceed<br/><br/>created by the requester<br/>typically a mapper)
+    end
+
+    subgraph executing_state["status: executing"]
+      executing(the command is under execution<br/>the content is the same as for init)
+    end
+
+    subgraph successful_state["status: successful"]
+      successful(the command has been successful<br/>some outcome might be returned to the requester)
+    end
+
+    subgraph failed_state["status: failed"]
+      failed(the command has failed<br/>the reason is given to the requester)
+    end
+
+    subgraph done_state[done]
+      done(finally cleared by the requester)
+    end
+```
+
+
+### Capability declaration
+
+On start, a service that implements an operation for a device must:
+
+- know the MQTT root prefix (per default: `te`),
+- know the entity topic identifier of the target device (e.g. `device/child001//`),
+- know the well-known name of the operation (e.g. `software-update`),
+- forge a capability message describing the operation support (e.g. `{ "supportedTypes": ["apt", "docker"] }`)
+- publish this capability message, with a retained flag, on the target command metadata topic (e.g. `te/device/child001///cmd/software-update`)
+
+### Command execution
+
+To request the execution of a command, a client must:
+- know the MQTT root prefix (per default: `te`),
+- know the entity topic identifier of the target device (e.g. `device/child001//`),
+- know the well-known name of the operation (e.g. `software-update`),
+- assign a unique id to its command (say `c8y-1234`)
+- use these target and command identifiers to forge the topic name for its request
+  (e.g. `te/device/child001///cmd/software-update/c8y-1234`)
+- describe the request in a JSON message as defined by the operation API
+- publish, as a retained message, the request on its specific topic with a `"status"` set to `"init"`
+- await on the same topic for a response message with a `"status"` set to `"executing"`, `"successful"` or `"failed"`
+- ignore any message with an unknown status as the agent is free to use intermediate states
+- finally, clear the command topic by sending an empty retained message
+
+To implement an operation on behalf of a device, a service must:
+- know the MQTT root prefix (per default: `te`),
+- know the entity topic identifier of the target device (e.g. `device/child001//`),
+- know the well-known name of the operation (e.g. `software-update`),
+- subscribe to any command requests of this type and the target device (i.e. `te/device/child001///cmd/software-update/+`)
+- react to a well-formed requests by publishing the updated status when the step is done.
```

Comment:
Again, not worth repeating I feel.

Reply:
Indeed, I would not repeat that all and every operation. However, this must be said once.